src/Mvc/Mvc.Core/src/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -80,14 +81,15 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
                 throw new ArgumentNullException(nameof(readerFactory));
             }
 
-            _formatters = formatters;
-            _readerFactory = readerFactory.CreateReader;
-
-            if (loggerFactory != null)
+            if (loggerFactory is null)
```

Comment:
Another implementation change. We previously allowed a null logger factory, primarily to support the other ctor. Seems reasonable to require it instead.